OMRON NJ/NX Series Ethernet/IP initial support.

j2mcv opened the pull request:
Proposed changes to add support for OMRON NJ/NX PLCs.

Added a new cpu (AB_PLC_OMRON_NJNX) under the ab_eip protocol.

Uses the existing code paths for MLGX800, I've added a new tag_vtable_t to handle cases where cpu doesn't support partial data and extra parameters to the build_[read|write]_request functions.

I also had to set to zero the ip address length in match_ip_addr_segment to make in work in Windows under MSVC.

Commit:
OMRON NJ/NX Series Ethernet/IP initial support.

(cherry picked from commit c3188f7bc9ec2db66ee7eb5950a7b87a55b9338f)

kyle-github commented:
Hi @musiguello,

I am right in the middle of doing some major refactoring of the data accessor functions, so I am not going to merge this right away.  I have not forgotten it!

j2mcv commented:
Hi @kyle-github,

thanks!

> Hi @musiguello,
> 
> I am right in the middle of doing some major refactoring of the data accessor functions, so I am not going to merge this right away. I have not forgotten it!

j2mcv committed:
Return an error if a path is not defined on tag creation for omron plcs.

kyle-github commented:
Hi Jorge,

Thanks so much for this work!   I will make some tweaks to get it to build (I refactored the vtables when I moved around the data accessor functions).

Are you OK to license this under the same dual MPL 2.0/LGPL 2+ license as the core library?

j2mcv commented:
Yes, no problem.

Thanks!

El dom., 26 jul. 2020 1:25, Kyle Hayes <notifications@github.com> escribió:

> Hi Jorge,
>
> Thanks so much for this work! I will make some tweaks to get it to build
> (I refactored the vtables when I moved around the data accessor functions).
>
> Are you OK to license this under the same dual MPL 2.0/LGPL 2+ license as
> the core library?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/libplctag/libplctag/pull/153#issuecomment-663916259>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ADZ6FLTZDWE2C2HX3NNTP5LR5NSV3ANCNFSM4ONC6ISA>
> .
>

kyle-github commented:
Thanks!
